src/diggler/bot.d

```diff
@@ -170,10 +177,11 @@ final class Bot
 		this.preferredNick = enforce(conf.nickName, "must specify nick name", file, line);
 		this._userName = enforce(conf.userName, "must specify user name", file, line);
 		this._realName = enforce(conf.realName, "must specify the real name field", file, line);
+		this.help = help;
 
 		this._commandQueue = new CommandQueue();
 
-		registerCommands(new DefaultCommands(this));
+		if(help != HelpCommand.none) registerCommands(new DefaultCommands(this));
```

Comment:
While you're at it, please put the if-body on its own line (you don't have to add braces)